### DISCUSSION TITLE
Some channels are not receiving messages after the Upgrade from 3.8.1 to 4.4.2

### DISCUSSION THREAD

**Discussion by Jaysarun** [nextgenhealthcare/connect#6179] [Q&A]
Recently we have upgraded our mirth connect from 3.8.1 to 4.4.2. After that some of the channels are not getting the messages flow into our system and some of them are working fine. To make sure of this issue again we downgraded to 3.8.1 version, there messages are flowing in. We check the codes, channels configuration they all are same. Could you please guide me on this one?

**Comment by jonbartels** [comment]
What are the source connector types of the affected channels?

What do you see in `mirth.log`?

What does a packet capture show?

**Reply by Jaysarun** [reply]
Hi @jonbartels,

1. Every Channels source connector types are TCP Listener.

2. The Mirth Log is not showing anything it was only showing the details of the channels that are getting connected and receiving the messages. No error messages about the other channels. I would like to share the logs with but since we put back the old version we didn't have the new version logs.

3. The packet capture shows 0.

**Reply by pacmano1** [reply]
Try a _clean_ install rather than an upgrade.   There are some reports of upgrades not cleaning up old libraries. 

This isn't really a mirth question in my opinion, it's a fundamental "how to I manage and confirm appropriate upgrades" for any piece of software.

You also provided almost no details on your configuration.   Operating system?  Java version and distribution?  Backend DB?  etc.  @jonbartels mentioned mirth.log.   I would also wonder what netstat shows.   I'd want to see mirth.properties.  I'd want to know of your host is multi-homed and perhaps mirth was listening on the wrong interface.  

You have posted here enough to know to provide comprehensive information not "it's broken".

**Comment by Jaysarun** [comment]
Yes, we did a clean installation by cleaning up the old libraries. But still we are facing this issue.

**Reply by pacmano1** [reply]
A clean install would remove _everything_ in any mirth related directory, not just "libraries" and you would merge back into mirth.properties from the older version.   You would want to restore the server id file also or you lose channel stats.

**Comment by Jaysarun** [comment]
Noted @pacmano1 , I will try it and let you know. Thanks!

**Reply by cranicaudal** [reply]
I do have had the same Problem with Channel getting Data via TCP/IP. Solved it by setting the Listener Interface from default to the IP of the Mirth Server Interface.

May that help you

**Reply by pacmano1** [reply]
That is a good point.

I assume the original poster confirmed the listener by checking netstat.
